src/forms/spaceEditSchema.ts

```diff
@@ -19,12 +19,22 @@ const roomImageUrlSchema = Yup.string().required(
   `${ROOM_TAXON.capital} icon is required`
 );
 
+const roomSchema = Yup.object().shape<RoomSchemaShape>({
+  title: createNameSchema({ name: "Title", withMin: true }),
+  image_url: roomImageUrlSchema,
+});
+
+const roomSchemaWithUrl = { ...roomSchema, url: roomUrlSchema };
```

Comment:
This can be a conditional inside a factory. You can check https://github.com/sparkletown/sparkle/blob/staging/src/forms/factory/createNameSchema.ts

Reply:
redone this in a factory fashion c72f598